src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/SearchTree.cs

```diff
@@ -117,10 +129,51 @@ public IReadOnlyList<string> GetConfiguredPackageSources(string term)
 
             if (i == term.Length && currentNode.PackageSources != null)
             {
-                return currentNode.PackageSources;
+                return currentNode;
+            }
+
+            return longestMatchingPrefixNode;
+        }
+
+        private string SearchPatternByTerm(string term)
```

Comment:
Wanted to find matching patterns, and return the pattern. Using a `StringBuilder` and same algorithm as `GetConfiguredPackageSources`. 
/cc @erdembayar - any concerns?